core/src/main/java/io/questdb/PropServerConfiguration.java

```diff
@@ -358,346 +364,438 @@ public PropServerConfiguration(
+                }
+                this.lineTcpWriterQueueCapacity = getInt(properties, env, "line.tcp.writer.queue.capacity", 128);
+                this.lineTcpWriterWorkerCount = getInt(properties, env, "line.tcp.writer.worker.count", 1);
+                cpuUsed += this.lineTcpWriterWorkerCount;
+                this.lineTcpWriterWorkerAffinity = getAffinity(properties, env, "line.tcp.writer.worker.affinity", lineTcpWriterWorkerCount);
+                this.lineTcpWriterWorkerPoolHaltOnError = getBoolean(properties, env, "line.tcp.writer.halt.on.error", false);
+                this.lineTcpWriterWorkerYieldThreshold = getLong(properties, env, "line.tcp.writer.worker.yield.threshold", 10);
+                this.lineTcpWriterWorkerSleepThreshold = getLong(properties, env, "line.tcp.writer.worker.sleep.threshold", 10000);
+                this.symbolCacheWaitUsBeforeReload = getLong(properties, env, "line.tcp.symbol.cache.wait.us.before.reload", 500_000);
+
+                int ilpTcpWorkerCount;
+                if (cpuAvailable < 9) {
+                    ilpTcpWorkerCount = 0;
+                } else if (cpuAvailable < 17) {
+                    ilpTcpWorkerCount = 2;
+                } else {
+                    ilpTcpWorkerCount = 6;
+                }
+                this.lineTcpIOWorkerCount = getInt(properties, env, "line.tcp.io.worker.count", ilpTcpWorkerCount);
+                cpuUsed += this.lineTcpIOWorkerCount;
+                this.lineTcpIOWorkerAffinity = getAffinity(properties, env, "line.tcp.io.worker.affinity", lineTcpIOWorkerCount);
+                this.lineTcpIOWorkerPoolHaltOnError = getBoolean(properties, env, "line.tcp.io.halt.on.error", false);
+                this.lineTcpIOWorkerYieldThreshold = getLong(properties, env, "line.tcp.io.worker.yield.threshold", 10);
+                this.lineTcpIOWorkerSleepThreshold = getLong(properties, env, "line.tcp.io.worker.sleep.threshold", 10000);
+                this.lineTcpNUpdatesPerLoadRebalance = getInt(properties, env, "line.tcp.n.updates.per.load.balance", 10_000_000);
+                this.lineTcpMaxLoadRatio = getDouble(properties, env, "line.tcp.max.load.ratio", 1.9);
+                this.lineTcpMaintenanceInterval = getInt(properties, env, "line.tcp.maintenance.job.interval", 30_000);
+                this.lineTcpAuthDbPath = getString(properties, env, "line.tcp.auth.db.path", null);
+                String defaultPartitionByProperty = getString(properties, env, "line.tcp.default.partition.by", "DAY");
+                this.lineDefaultPartitionBy = PartitionBy.fromString(defaultPartitionByProperty);
+                if (this.lineDefaultPartitionBy == -1) {
+                    log.info().$("invalid partition by ").$(defaultPartitionByProperty).$("), will use DAY").$();
+                    this.lineDefaultPartitionBy = PartitionBy.DAY;
+                }
+                if (null != lineTcpAuthDbPath) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of != on line 756 to == check before setting the field value on line 757 or verify if the check is needed at all.